src/metabase/lib/drill_thru/column_filter.cljc

```diff
@@ -23,10 +29,26 @@
                        (-> (lib.filter.operator/filter-operators column)
                            first
                            (assoc :lib/type :operator/filter)))]
-      {:lib/type   :metabase.lib.drill-thru/drill-thru
-       :type       :drill-thru/column-filter
-       :column     column
-       :initial-op initial-op})))
+
+      (merge
+        {:lib/type   :metabase.lib.drill-thru/drill-thru
+         :type       :drill-thru/column-filter
+         :column     column
+         :initial-op initial-op}
+        ;; When the column we would be filtering on is an aggregation, it can't be filtered without adding a stage.
+        (if (= (:lib/source column) :source/aggregations)
+          (let [next-stage    (->> (lib.util/canonical-stage-index query stage-number)
+                                   (lib.util/next-stage-number query))
+                ;; next-stage is nil if there's no stage after the current one, so add one and bump the stage-number.
+                base          (if next-stage
+                                {:query        query
+                                 :stage-number next-stage}
+                                {:query        (lib.stage/append-stage query)
+                                 :stage-number -1})]
+            (assoc base :column (->> (lib.filter/filterable-columns (:query base) (:stage-number base))
```

Comment:
@ranquild Here's the code where the returned column is coming from `filterable-columns`.